client/src/reducers/socket.js

```diff
@@ -9,11 +9,17 @@ export default function SocketReducer(state = {
 			message: action.message
 		};
 	case 'SOCKET_NEW_TICKET': {
-		const newTicket = action.ticket;
-		newTicket.timestamp = new Date(newTicket.timestamp);
+		const { timeFiled } = action.newTicket;
+		const ticketWithDate = { ...action.newTicket, timeFiled: new Date(timeFiled) };
 		return {
 			...state,
-			tickets: [...state.tickets, newTicket]
+			tickets: [...state.tickets, ticketWithDate]
+		};
+	}
+	case 'SOCKET_ERROR': {
+		return {
+			...state,
+			errorMsg: { ...action.error }
```

Comment:
Does the `errorMsg` ever get cleared anywhere? The error might be temporary.

Reply:
It will not be cleared yet. I think we will have a PR in the future to handle these errors. Don't want this PR to get huge at this point.

Reply:
Let's get rid of the `SOCKET_ERROR` action altogether then, since it's no longer used in this PR.